rosbag2_transport/src/rosbag2_transport/qos.cpp

```diff
@@ -21,6 +21,51 @@
 
 namespace
 {
+
+// Foxy compatibility - copied from Galactic version of rmw/time.c
+typedef rcutils_duration_value_t rmw_duration_t;
+// we can't use the Galactic version of this constant because Foxy RMWs won't understand it
+static const rmw_time_t RMW_DURATION_INFINITE {0LL, 0LL};
+
+
+rmw_duration_t
```

Comment:
Other functions (see line 97) have the function type inline with the function name. Should this follow that style?